### ISSUE TITLE
HTML list elements in sidenote do not appear correctly

### ISSUE THREAD

**Issue by statzhero** [edwardtufte/tufte-css#93]
I tried to include HTML list elements in a sidenote, but that did not work. By this I mean the list did not appear in the sidenote but all over the place. Again, apologies if I'm missing something.

**Comment by crmackay** [comment]
so marginnotes and sidenotes, as they are implemented now, are `<span>` elements inside paragraph `<p>` elements. 

Only inline elements are allowed inside paragraphs per the HTML standard. 

`<ol>` and `<ul>` are themselves block level elements, so they cannot go inside `<span>` or `<p>` elements.

Can you provide an example of what you are trying to do and why? Maybe we can come up with a solution.

**Comment by statzhero** [comment]
Ah that explains it. I think I can live without it. 😃 

Maybe something for the future to make sidenotes and marginnotes more flexible... but I'm afraid I wouldn't know how to do it.

**Comment by daveliepmann** [comment]
Per reasoning from @crmackay I'm closing this, but we should keep it in mind for future expansions of functionality (that address shortcomings in HTML itself :D)

**Comment by achmizs** [comment]
Putting block elements inside spans is very possible and also very easy to make it work properly via CSS. I played around with the CSS for a couple of minutes and was able to get this:

[https://dl.dropboxusercontent.com/s/wal5xf7jjr26189/Screen%20Shot%202020-08-07%20at%204.02.40%20PM.png?dl=0l](https://dl.dropboxusercontent.com/s/wal5xf7jjr26189/Screen%20Shot%202020-08-07%20at%204.02.40%20PM.png?dl=0l)

The HTML for this screenshot is simply three `<p>` elements within the `span.sidenote`, as you would expect.

In short, making this work requires only some minor tweaks to the style sheet.

**Comment by benisjammin** [comment]
What CSS tweaks did you make that?

**Comment by achmizs** [comment]
First, I edited the HTML to see what it would look like with `<p>` tags in the sidenote `<span>` but no CSS changes. The result was this:

https://dl.dropboxusercontent.com/s/9gwyus36jbw2mbq/Screen%20Shot%202020-08-09%20at%205.14.55%20PM.png?dl=0

The problem is twofold. First, the font size is wrong (as is the line height and the paragraph spacing)[1]. Second, the sidenote number is improperly positioned.

The first problem is fixed like so:

```
.sidenote p {
    font-size: 1.1rem;
    line-height: 1.3;
    margin: 1em 0;
}
.sidenote p:first-child {
    margin-top: 0;
}
.sidenote p:last-child {
    margin-bottom: 0;
}
```

It is possible (indeed, probable) that similar adjustments will need to be made for other block elements (like `<ul>` and `<ol>`) when they appear within sidenotes.

The second problem is fixed like so:

```
.sidenote::before {
    position: absolute;
    right: calc(100% + 0.5em);
}
```

The result of these changes looks like this:

https://dl.dropboxusercontent.com/s/y5r14qrmj4tmlkx/Screen%20Shot%202020-08-09%20at%205.24.22%20PM.png?dl=0

---

[1] Note that this problem is caused by specifying the given properties in `rem` units, which is generally inadvisable for precisely reasons like this.
